lib/charms/opensearch/v0/opensearch_plugins.py

```diff
@@ -332,6 +332,41 @@ class PluginState(BaseStrEnum):
     WAITING_FOR_UPGRADE = "waiting-for-upgrade"
 
 
+class OpenSearchPluginRelationsHandler(ABC):
+    """Implements the relation manager for each plugin.
+
+    Plugins may have one or more relations tied to them. This abstract class
```

Comment:
I guess this is the case for the backup plugin with Azure or AWS S3, right?

Reply:
Yes